### PR TITLE
Upload files to discussion threads

### PR DESCRIPTION
**Pull request by adam-incuna**
Allow users to upload a file as a comment in a discussion thread.

This involves:
* Adding a new subclass of `BaseComment` called `FileComment` that holds a file.
* Adding a form that allows users to upload `FileComment`s.  The `AddComment` form has been turned into a superclass for this and `AddTextComment`, which fulfils `AddComment`'s old functionality.
* Adding a view to provide that form.  The view shared a lot of functionality with `DiscussionThread`, so I pulled out the common elements into a base class called `CommentPostView`.  This will also be convenient for other types of comment postings, and allows people to easily add new upload views.
* New templates and updates to the discussion thread template.

@incuna/backend review please? :)

### CODE REVIEW

**Comment by adam-incuna** [comment]
Travis says I've been cushioned by Django 1.7 niceness.  Working on it.

File: groups/models.py [hunk 1]
```diff
@@ -143,3 +143,10 @@ class TextComment(BaseComment):
 
     class Meta(BaseComment.Meta):
         pass
+
+
+class FileComment(BaseComment):
+    file = models.FileField()
+
+    class Meta(BaseComment.Meta):
+        pass
```
Comment (KevinEtienne) [inline groups/models.py]: I think you don't need the `Meta` class as it would be inherited from the parent `BaseComment`

File: groups/forms.py [hunk 2]
```diff
@@ -7,23 +7,49 @@
...
+        helper = super(AddFileComment, self).build_helper()
+        helper.layout = Layout(
+            'file',
+            FormActions(
+                Submit('comment-submit', 'Upload this file'),
+            ),
+        )
+        return helper
+
+    class Meta(BaseAddCommentForm.Meta):
```
Comment (KevinEtienne) [inline groups/forms.py]: I think you don't need to inherit from `BaseAddCommentForm` as there's no Meta on it. Same apply for `AddTextComment`

**Comment by adam-incuna** [comment]
@KevinEtienne done, thanks! :)

**Comment by KevinEtienne** [comment]
Looks good to me, I think it needs a some docstrings on form and models and good to merge

**Comment by adam-incuna** [comment]
On it

**Comment by adam-incuna** [comment]
@KevinEtienne cheers, done!

**Comment by adam-incuna** [comment]
thanks! :)

**Pull request merged**